### ISSUE TITLE
If CGO is enabled, cross compilation fails with misdirecting error message.

### ISSUE THREAD

**Issue by jaredfolkins** [apex/up#356]
## Prerequisites

* [x] I searched to see if the issue already exists.
* [x] I am running the latest version. (`up upgrade`) 

```0.2.6```

* [x] I inspected the verbose debug output with the `-v, --verbose` flag.

```
Error: building: build hook: "GOOS=linux GOARCH=amd64 go build -o server *.go": # runtime/cgo
ld: unknown option: --build-id=none
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

## Description

If environment variable ```export CGO_ENABLED=1``` is set, Up's will fail with a unhelpful error message. The reason is that Go tries to compile net/http using the CGO resolver.

"By default the pure Go resolver is used, because a blocked DNS request consumes only a goroutine, while a blocked C call consumes an operating system thread. When cgo is available, the cgo-based resolver is used instead under a variety of conditions"

https://golang.org/pkg/net/

## Steps to Reproduce

[1] ```$ export CGO_ENABLED=1```

[2] clone the golang Up api example

[3] run ```$ up``` in project directory

## Love Up?

❤️

## Solution

Check if CGO is enabled at compile time and warn? Fail?

**Comment by skabbes** [comment]
I don't know if you're looking for a workaround or just a helpful message - I think we should have both :)

Regarding the workaround, I think you'll need to define a custom build & clean hook that possibly builds with docker.

**Comment by jaredfolkins** [comment]
> Regarding the workaround, I think you'll need to define a custom build & clean hook that possibly builds with docker.

Simply assert ```export CGO_ENABLED=0``` with your build and clean for a workaround. **No docker required.**

From a product on-boarding perspective when using Up for the first time this was a bad experience. Figured it'd be best to remedy that so that it isn't a turn off for others. It sounds like you agree. But if not, no worries if you close it.

**Comment by skabbes** [comment]
No, I definitely agree.  I was just asking because some projects _need_ CGO enabled to be able to compile in the first place.  I wasn't sure if you fell into that category.

Feel free to add a PR, it _could_ be done [here](https://github.com/apex/up/blob/master/config.go#L314)  or it could also be asserted at build time.

**Comment by tj** [comment]
sounds reasonable to force `0` it in the hook by default :D
